spec/compiler/codegen/generic_class_spec.cr

```diff
@@ -96,6 +96,53 @@ describe "Code gen: generic class type" do
       )).to_i.should eq(1)
   end
 
+  it "runs generic instance var initializers in superclass's metaclass context (#4753)" do
+    run(%(
+      class Bar(T)
+        def x
+          {% if T == Int32 %} 1 {% else %} 2 {% end %}
+        end
+      end
+
+      class FooBase(T)
+        @bar = Bar(T).new
+
+        def bar
+          @bar
+        end
+      end
+
+      class Foo(T) < FooBase(T)
+      end
+
+      Foo(Int32).new.bar.x
+      )).to_i.should eq(1)
+  end
+
+  it "runs generic instance var initializers in superclass's metaclass context (2) (#6482)" do
+    run(%(
+      class Bar(T)
+        def x
+          # TODO: devirtualize T
+          {% if T <= FooBase(Int32) && T >= FooBase(Int32) %} 1 {% else %} 2 {% end %}
```

Comment:
This is a special case of #10730. Note that `TypeNode#<=` and the other comparison methods do [devirtualize the types](https://github.com/crystal-lang/crystal/blob/a2c632e5d6f0c4b988d6070fab81614040683645/src/compiler/crystal/macros/methods.cr#L1702-L1703).